Source/EasyNetQ/PersistentConnection.cs

```diff
@@ -54,18 +56,29 @@ public class PersistentConnection : IPersistentConnection
             this.eventBus = eventBus;
         }
 
-        /// <inheritdoc />
-        public IModel CreateModel()
+        public void Connect()
+        {
+            initializedConnection = Initialize();
+        }
+
+        private IAutorecoveringConnection Initialize()
         {
+
             var connection = initializedConnection;
             if (connection == null)
                 lock (mutex)
-                    connection = initializedConnection ??= Connect();
+                    connection = initializedConnection ??= ConnectInternal();
```

Comment:
`ConnectInternal()` creates the connection and fires the connected event however `IsConnected` is still false at the time since it will be set when `ConnectInternal()` returns or am I missing something?